diff --git a/CMakeLists.txt b/CMakeLists.txt
index 36f6587..a799615 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -678,7 +678,7 @@ endif()
 if(ENABLE_GLIB)
   add_subdirectory(glib)
 endif()
-add_subdirectory(test)
+#add_subdirectory(test)
 if(ENABLE_QT5)
   add_subdirectory(qt5)
 endif()
diff --git a/cpp/CMakeLists.txt b/cpp/CMakeLists.txt
index 4d8bc7b..2d96a7c 100644
--- a/cpp/CMakeLists.txt
+++ b/cpp/CMakeLists.txt
@@ -6,7 +6,7 @@ include_directories(
 
 configure_file(poppler-version.h.in ${CMAKE_CURRENT_BINARY_DIR}/poppler-version.h @ONLY)
 
-add_subdirectory(tests)
+#add_subdirectory(tests)
 
 set(poppler_cpp_SRCS
   poppler-destination.cpp
diff --git a/qt5/CMakeLists.txt b/qt5/CMakeLists.txt
index 624ea64..3b94059 100644
--- a/qt5/CMakeLists.txt
+++ b/qt5/CMakeLists.txt
@@ -6,5 +6,5 @@ if(ENABLE_QT_STRICT_ITERATORS)
 endif()
 
 add_subdirectory(src)
-add_subdirectory(tests)
-add_subdirectory(demos)
+#add_subdirectory(tests)
+#add_subdirectory(demos)
