From a200efdbd105bf1c03b04e037d59faac70b83b4e Mon Sep 17 00:00:00 2001
From: Dmitry Kazakov <dimula73@gmail.com>
Date: Fri, 7 Apr 2023 13:50:12 +0200
Subject: [PATCH] Fix library filename on Windows

---
 win32/ogg.def | 1 -
 1 file changed, 1 deletion(-)

diff --git a/win32/ogg.def b/win32/ogg.def
index 030b644..614031b 100644
--- a/win32/ogg.def
+++ b/win32/ogg.def
@@ -2,7 +2,6 @@
 ; ogg.def
 ; List of exported functions for Windows builds.
 ;
-LIBRARY ogg
 EXPORTS
 ;
 oggpack_writeinit
-- 
2.38.1.windows.1

