From 1cd87709f9ed56c5b3a51fb95b7fd3230629c7bd Mon Sep 17 00:00:00 2001
From: Dmitry Kazakov <dimula73@gmail.com>
Date: Wed, 31 May 2023 16:00:54 +0300
Subject: [PATCH] [ffmpeg] Be more forceful when setting PIC on x86_32

---
 meson.build | 1 +
 1 file changed, 1 insertion(+)

diff --git a/meson.build b/meson.build
index 9942b2d120..a75ef4deb0 100644
--- a/meson.build
+++ b/meson.build
@@ -1506,6 +1506,7 @@ if cpu == 'x86'
   conf.set('x86', 1)
   conf.set('i686', 1)
   conf.set('x86_32', 1)
+  conf.set10('pic', true)
 elif cpu == 'x86_64'
   arch = 'x86'
   conf.set('x86', 1)
-- 
2.25.1

