From 3881bc2fa44806b4f05475b694caa25695dbed09 Mon Sep 17 00:00:00 2001
From: "L. E. Segovia" <amy@amyspark.me>
Date: Fri, 25 Nov 2022 14:59:14 -0300
Subject: [PATCH 02/11] theora: Add support for hidden frame dumping flag

---
 lib/meson.build | 2 +-
 meson.build     | 2 ++
 2 files changed, 3 insertions(+), 1 deletion(-)

diff --git a/lib/meson.build b/lib/meson.build
index 12e6823..965831c 100644
--- a/lib/meson.build
+++ b/lib/meson.build
@@ -214,7 +214,7 @@ theoraenc_dep = declare_dependency(
 libtheora = library(
   'theora',
   encoder_sources + decoder_sources,
-  dependencies: [m_dep, ogg_dep, config_dep],
+  dependencies: [m_dep, libpng_dep, ogg_dep, config_dep],
   include_directories: incdir,
   c_args: compile_args,
   link_args: theora_version_flag,
diff --git a/meson.build b/meson.build
index c5ba8ca..a7515a8 100644
--- a/meson.build
+++ b/meson.build
@@ -30,6 +30,8 @@ endif
 
 ogg_dep = dependency('ogg', version: '>= 1.1', required: true)
 
+libpng_dep = dependency('libpng', required: false)
+
 asm = not get_option('asm').disabled()
 if asm
   if host_machine.cpu_family() == 'x86_64'
-- 
2.37.1.windows.1

