From 39cc6fca20e9f59065a698bd83c603820c3a73ee Mon Sep 17 00:00:00 2001
From: "L. E. Segovia" <amy@amyspark.me>
Date: Sat, 26 Nov 2022 22:08:00 +0000
Subject: [PATCH 03/11] theora: Add missing libtheora module definition file

---
 lib/meson.build | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/lib/meson.build b/lib/meson.build
index 965831c..abb7faf 100644
--- a/lib/meson.build
+++ b/lib/meson.build
@@ -218,7 +218,7 @@ libtheora = library(
   include_directories: incdir,
   c_args: compile_args,
   link_args: theora_version_flag,
-  # vs_module_defs: 'theora.def', # Missing in the official tarballs
+  vs_module_defs: 'theora.def', # Missing in the official tarballs
   install: true,
 )
 
-- 
2.37.1.windows.1

