From 385f5c8c95a1de06ee1742eae0228d63e1a59dac Mon Sep 17 00:00:00 2001
From: "L. E. Segovia" <amy@amyspark.me>
Date: Mon, 21 Nov 2022 10:30:59 -0300
Subject: [PATCH 9/9] meson: Enable icudata for MSVC

---
 source/data/meson.build | 28 +++++++++++++---------------
 1 file changed, 13 insertions(+), 15 deletions(-)

diff --git a/source/data/meson.build b/source/data/meson.build
index 0cbb9b7..9e020d9 100644
--- a/source/data/meson.build
+++ b/source/data/meson.build
@@ -68,21 +68,19 @@ if host_machine.system() == 'windows'
   )
 endif
 
-if cpp.get_id() != 'msvc'
-  icudata = library(
-    icudata_name,
-    sources,
-    include_directories: incdir,
-    version: U_ICU_VERSION,
-    vs_module_defs: icudata_exports,
-    install: true,
-  )
+icudata = library(
+  icudata_name,
+  sources,
+  include_directories: incdir,
+  version: U_ICU_VERSION,
+  vs_module_defs: icudata_exports,
+  install: true,
+)
 
-  icudata_dep = declare_dependency(
-    compile_args: usage_args,
-  )
+icudata_dep = declare_dependency(
+  compile_args: usage_args,
+)
 
-  if meson.version().version_compare('>=0.54.0')
-    meson.override_dependency('icu-data', icudata_dep)
-  endif
+if meson.version().version_compare('>=0.54.0')
+  meson.override_dependency('icu-data', icudata_dep)
 endif
-- 
2.37.1.windows.1

