From 1b4d4a5c1754ba12b21866ba86548919c191c855 Mon Sep 17 00:00:00 2001
From: "L. E. Segovia" <amy@amyspark.me>
Date: Thu, 17 Nov 2022 01:01:47 +0000
Subject: [PATCH 12/18] meson: fix compatibility with 0.49.0

---
 source/tools/toolutil/meson.build | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/source/tools/toolutil/meson.build b/source/tools/toolutil/meson.build
index 636f654..ed0e279 100644
--- a/source/tools/toolutil/meson.build
+++ b/source/tools/toolutil/meson.build
@@ -45,5 +45,5 @@ toolutil_lib = library(
 
 toolutil_dep = declare_dependency(
   link_with: toolutil_lib,
-  include_directories: '.',
+  include_directories: include_directories('.'),
 )
-- 
2.37.1.windows.1

