From 65089e1c01dcab9ccfce36ef66e44aabc91fef40 Mon Sep 17 00:00:00 2001
From: Dmitry Kazakov <dimula73@gmail.com>
Date: Fri, 25 Nov 2022 09:31:29 +0300
Subject: [PATCH] Fix compilation of KCoreAddons with Qt 5.15.7

---
 src/lib/randomness/krandom.cpp | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/lib/randomness/krandom.cpp b/src/lib/randomness/krandom.cpp
index 85cb5a3..a7170be 100644
--- a/src/lib/randomness/krandom.cpp
+++ b/src/lib/randomness/krandom.cpp
@@ -73,7 +73,7 @@ QString KRandom::randomString(int length)
         if (r > 90) {
             r += 6;
         }
-        str[i++] =  char(r);
+        str[i++] = QChar(r);
         // so what if I work backwards?
     }
     return str;
-- 
2.38.1.windows.1

