From a321bbe597e6b26fbdc46c392fe31d74af16a38a Mon Sep 17 00:00:00 2001
From: Sharaf Zaman <shzam@sdf.org>
Date: Sun, 21 May 2023 22:42:27 +1000
Subject: [PATCH] :ANDROID]: Find dependencies of manually specified libraries

---
 toolchain/specifydependencies.cmake | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/toolchain/specifydependencies.cmake b/toolchain/specifydependencies.cmake
index 93ea2635..23704e2b 100644
--- a/toolchain/specifydependencies.cmake
+++ b/toolchain/specifydependencies.cmake
@@ -57,6 +57,8 @@ if (ANDROID_EXTRA_LIBS)
         else()
             message(STATUS "manually specified extra library: " ${extralib})
             list(APPEND extralibs ${extralib})
+            # Now, find dependencies of manually added library.
+            list_dependencies(${extralib} extralibs)
         endif()
     endforeach()
 endif()
-- 
2.39.2

