From 311ecc277a4cc2cc4c28c836a25bbcb3331f940b Mon Sep 17 00:00:00 2001
From: Dmitry Kazakov <dimula73@gmail.com>
Date: Fri, 25 Nov 2022 09:38:22 +0300
Subject: [PATCH] Fix compilation of KItemViews with Qt 5.15.7

---
 src/kwidgetitemdelegatepool_p.h | 1 +
 1 file changed, 1 insertion(+)

diff --git a/src/kwidgetitemdelegatepool_p.h b/src/kwidgetitemdelegatepool_p.h
index 206ee0e..9d0147b 100644
--- a/src/kwidgetitemdelegatepool_p.h
+++ b/src/kwidgetitemdelegatepool_p.h
@@ -24,6 +24,7 @@
 
 #include <QHash>
 #include <QList>
+#include <QPersistentModelIndex>
 
 class QWidget;
 class QStyleOptionViewItem;
-- 
2.38.1.windows.1

