From 769af8658411ce9fd785c56754d8d9d10a1e61da Mon Sep 17 00:00:00 2001
From: Sharaf Zaman <shzam@sdf.org>
Date: Mon, 5 Dec 2022 11:16:30 +0000
Subject: [PATCH] Android: Skip setting up fastlane

---
 toolchain/ECMAndroidDeployQt.cmake | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/toolchain/ECMAndroidDeployQt.cmake b/toolchain/ECMAndroidDeployQt.cmake
index 65b68c67..a7a467da 100644
--- a/toolchain/ECMAndroidDeployQt.cmake
+++ b/toolchain/ECMAndroidDeployQt.cmake
@@ -125,5 +125,5 @@ function(ecm_androiddeployqt QTANDROID_EXPORTED_TARGET ECM_ADDITIONAL_FIND_ROOT_
         COMMAND adb install -r "${ANDROID_APK_OUTPUT_DIR}/${QTANDROID_EXPORTED_TARGET}-${CMAKE_ANDROID_ARCH_ABI}.apk"
     )
     add_dependencies(create-apk ${CREATEAPK_TARGET_NAME})
-    add_dependencies(${CREATEAPK_TARGET_NAME} create-fastlane)
+    # add_dependencies(${CREATEAPK_TARGET_NAME} create-fastlane)
 endfunction()
-- 
2.38.1

