diff --git a/src/desktoptojson/CMakeLists.txt b/src/desktoptojson/CMakeLists.txt
index dd668a4..424d14c 100644
--- a/src/desktoptojson/CMakeLists.txt
+++ b/src/desktoptojson/CMakeLists.txt
@@ -11,5 +11,5 @@ ecm_mark_nongui_executable(desktoptojson)
 
 target_link_libraries(desktoptojson Qt5::Core)
 target_compile_definitions(desktoptojson PRIVATE BUILDING_DESKTOPTOJSON_TOOL=1)
-
+set_target_properties(desktoptojson PROPERTIES INSTALL_RPATH "${CMAKE_INSTALL_PREFIX}/lib;${CMAKE_INSTALL_PREFIX}/bin")
 install(TARGETS desktoptojson EXPORT KF5CoreAddonsToolingTargets ${KF5_INSTALL_TARGETS_DEFAULT_ARGS})
