diff --git a/CMakeLists.txt b/CMakeLists.txt
index 20a6a29..01c7bec 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -40,11 +40,11 @@ if (UNIX)
     endif()
 endif()
 
-find_package(OpenEXR)
-set_package_properties(OpenEXR PROPERTIES
-    TYPE OPTIONAL
-    PURPOSE "Required for the QImage plugin for OpenEXR images"
-)
+# find_package(OpenEXR)
+# set_package_properties(OpenEXR PROPERTIES
+#     TYPE OPTIONAL
+#     PURPOSE "Required for the QImage plugin for OpenEXR images"
+# )
 add_definitions(-DQT_NO_FOREACH)
 # 050d00 (5.13) triggers a BIC in qimageiohandler.h, in Qt 5.13, so do not enable that until we can require 5.14
 # https://codereview.qt-project.org/c/qt/qtbase/+/279215
