commit e0ea4199bc4da359eb91ab51274785d17a4f2909
Author: Ralf Habacker <ralf.habacker@freenet.de>
Date:   Wed Sep 21 13:38:03 2016 +0200

    Windows compile fix.

diff --git a/src/lib/util/kuser_win.cpp b/src/lib/util/kuser_win.cpp
index 1d77f89..5c3fa45 100644
--- a/src/lib/util/kuser_win.cpp
+++ b/src/lib/util/kuser_win.cpp
@@ -853,7 +853,7 @@ static std::unique_ptr<char[]> queryProcessInformation(TOKEN_INFORMATION_CLASS t
     HANDLE _token;
     if (!OpenProcessToken(GetCurrentProcess(), TOKEN_QUERY, &_token)) {
         qWarning("Failed to get the token for the current process: %d", (int)GetLastError());
-        return false;
+        return nullptr;
     }
     ScopedHANDLE token(_token, handleCloser);
     // query required size
