diff --git a/src/imageformats/CMakeLists.txt b/src/imageformats/CMakeLists.txt
index 8a53bba..4d2f553 100644
--- a/src/imageformats/CMakeLists.txt
+++ b/src/imageformats/CMakeLists.txt
@@ -47,6 +47,9 @@ if(OpenEXR_FOUND)
     kimageformats_add_plugin(kimg_exr JSON "exr.json" SOURCES exr.cpp)
     target_link_libraries(kimg_exr OpenEXR::IlmImf)
     kde_target_enable_exceptions(kimg_exr PRIVATE)
+    if (WIN32)
+        add_definitions(-DOPENEXR_DLL)
+    endif()
 
     install(FILES exr.desktop DESTINATION ${KDE_INSTALL_KSERVICES5DIR}/qimageioplugins/)
 endif()
