diff --git a/srclib/stdio.in.h b/srclib/stdio.in.h
index 4a8aa55..9036d29 100644
--- a/srclib/stdio.in.h
+++ b/srclib/stdio.in.h
@@ -751,7 +751,7 @@ _GL_WARN_ON_USE (getline, "getline is unportable - "
    removed it.  */
 #undef gets
 #if HAVE_RAW_DECL_GETS && !defined __cplusplus
-_GL_WARN_ON_USE (gets, "gets is a security hole - use fgets instead");
+//_GL_WARN_ON_USE (gets, "gets is a security hole - use fgets instead");
 #endif
 
 #if @GNULIB_OBSTACK_PRINTF@ || @GNULIB_OBSTACK_PRINTF_POSIX@
