From de0c4dab234f8daebdea308be48f4d9f443e5b3a Mon Sep 17 00:00:00 2001
From: Halla Rempt <halla@valdyas.org>
Date: Wed, 30 Dec 2020 10:54:08 +0100
Subject: [PATCH] Remove chunk size limits

---
 src/gui/image/qpnghandler.cpp | 12 ++++++++++++
 1 file changed, 12 insertions(+)

diff --git a/src/gui/image/qpnghandler.cpp b/src/gui/image/qpnghandler.cpp
index e7e998ef..751ae4f0 100644
--- a/src/gui/image/qpnghandler.cpp
+++ b/src/gui/image/qpnghandler.cpp
@@ -542,6 +542,12 @@ bool QPngHandlerPrivate::readPngHeader()
 
     png_set_error_fn(png_ptr, 0, 0, qt_png_warning);
 
+#ifdef PNG_SET_USER_LIMITS_SUPPORTED
+      /* Remove the user limits, if any */
+      png_set_chunk_cache_max(png_ptr, 0);
+      png_set_chunk_malloc_max(png_ptr, 0);
+#endif    
+    
 #if defined(PNG_SET_OPTION_SUPPORTED) && defined(PNG_MAXIMUM_INFLATE_WINDOW)
     // Trade off a little bit of memory for better compatibility with existing images
     // Ref. "invalid distance too far back" explanation in libpng-manual.txt
@@ -831,6 +837,12 @@ bool QPNGImageWriter::writeImage(const QImage& image, volatile int compression_i
 
     png_set_error_fn(png_ptr, 0, 0, qt_png_warning);
 
+#ifdef PNG_SET_USER_LIMITS_SUPPORTED
+      /* Remove the user limits, if any */
+      png_set_chunk_cache_max(png_ptr, 0);
+      png_set_chunk_malloc_max(png_ptr, 0);
+#endif    
+    
     info_ptr = png_create_info_struct(png_ptr);
     if (!info_ptr) {
         png_destroy_write_struct(&png_ptr, 0);
-- 
2.25.1

