diff --git a/src/plugins/platforms/windows/qwindowstabletsupport.cpp b/src/plugins/platforms/windows/qwindowstabletsupport.cpp
index fa209f09c..d31998825 100644
--- a/src/plugins/platforms/windows/qwindowstabletsupport.cpp
+++ b/src/plugins/platforms/windows/qwindowstabletsupport.cpp
@@ -497,6 +497,11 @@ bool QWindowsTabletSupport::translateTabletPacketEvent()
             m_state = PenDown;
             m_mode = (mouseLocation - globalPosF).manhattanLength() > m_absoluteRange
                 ? MouseMode : PenMode;
+
+            // Krita doesn't support mouse mode. And this code may break
+            // normal painting, so we just disable it.
+            m_mode = PenMode;
+
             qCDebug(lcQpaTablet) << __FUNCTION__ << "mode=" << m_mode << "pen:"
                 << globalPosF << "mouse:" << mouseLocation;
         }
