From b5276ffe4cddb7e5b871a84438d6b8164803d4cc Mon Sep 17 00:00:00 2001
From: "L. E. Segovia" <amy@amyspark.me>
Date: Sun, 15 May 2022 14:56:21 +0000
Subject: [PATCH 131/131] Fix build with Android NDK 21.4

---
 src/tools/moc/generator.cpp | 1 +
 1 file changed, 1 insertion(+)

diff --git a/src/tools/moc/generator.cpp b/src/tools/moc/generator.cpp
index 12ffd6ae95..634ada7a65 100644
--- a/src/tools/moc/generator.cpp
+++ b/src/tools/moc/generator.cpp
@@ -42,6 +42,7 @@

 #include <math.h>
 #include <stdio.h>
+#include <limits>

 #include <private/qmetaobject_p.h> //for the flags.
 #include <private/qplugin_p.h> //for the flags.
--
2.35.1

