From 90eed4735d704a5236741ef48e8d17d11906d40c Mon Sep 17 00:00:00 2001
From: Alvin Wong <alvinhochun@gmail.com>
Date: Tue, 5 Apr 2022 18:49:33 +0800
Subject: [PATCH] HACK: Force glFinish in flushShared for ANGLE

---
 src/gui/opengl/qopenglfunctions.cpp | 31 +++++++++++++++++++++++++++++
 1 file changed, 31 insertions(+)

diff --git a/src/gui/opengl/qopenglfunctions.cpp b/src/gui/opengl/qopenglfunctions.cpp
index 92770cb55f..848e706d65 100644
--- a/src/gui/opengl/qopenglfunctions.cpp
+++ b/src/gui/opengl/qopenglfunctions.cpp
@@ -5072,6 +5072,37 @@ void QOpenGLExtensions::flushShared()
 {
     Q_D(QOpenGLExtensions);
 
+    if (!d->flushVendorChecked) {
+        static bool forceNoGlFinish = qEnvironmentVariableIntValue("KRITA_QT_NO_GL_FINISH_IN_FLUSH_SHARED") == 1;
+        if (forceNoGlFinish) {
+            d->flushVendorChecked = true;
+            d->flushIsSufficientToSyncContexts = true;
+        } else {
+#ifdef Q_OS_WIN
+            // HACK: Workaround QWidget composing too often and causing performance
+            // drop when using ANGLE by forcing a glFinish. This can be disabled
+            // by explicitly setting `KRITA_QT_NO_GL_FINISH_IN_FLUSH_SHARED=1`.
+            //
+            // FIXME: We ought to fix this issue with a more proper way...
+            //
+            // Note: GL_VENDOR on older ANGLE versions only contained "Google Inc.",
+            // but newer versions also included the native graphics vendor, e.g.
+            // "Google Inc. (Intel)". This changed the outcome of the check from
+            // above. In any case, we should explicitly check for ANGLE and its
+            // D3D backend.
+            const char *renderer = (const char *) glGetString(GL_RENDERER);
+            if (renderer && strstr(renderer, "ANGLE") && strstr(renderer, "Direct3D")) {
+                d->flushVendorChecked = true;
+                d->flushIsSufficientToSyncContexts = false;
+                static bool _callOnce = []() {
+                    qDebug() << "QOpenGLExtensions::flushShared() forcing glFinish for ANGLE";
+                    return true;
+                }();
+                Q_UNUSED(_callOnce)
+            }
+#endif
+        }
+    }
     if (!d->flushVendorChecked) {
         d->flushVendorChecked = true;
         // It is not quite clear if glFlush() is sufficient to synchronize access to
-- 
2.24.1.windows.2

