diff --git a/src/widgets/widgets/qabstractscrollarea.cpp b/src/widgets/widgets/qabstractscrollarea.cpp
index 598d173144..0be28bb941 100644
--- a/src/widgets/widgets/qabstractscrollarea.cpp
+++ b/src/widgets/widgets/qabstractscrollarea.cpp
@@ -1073,9 +1073,33 @@ bool QAbstractScrollArea::event(QEvent *e)
 #endif
         // ignore touch events in case they have been propagated from the viewport
     case QEvent::TouchBegin:
+        touchBeginTimestamp = static_cast<QTouchEvent*>(e)->timestamp();
     case QEvent::TouchUpdate:
+#ifdef Q_OS_ANDROID
+        return true;
+    case QEvent::TouchEnd: {
+        QTouchEvent *te = static_cast<QTouchEvent*>(e);
+        auto touchPoint = te->touchPoints().at(0);
+        const double maxDisDelta = 5.0;
+        const double maxTimestampDelta = 300;
+
+        QPointF start = te->touchPoints().at(0).startPos();
+        QPointF end = te->touchPoints().at(0).lastPos();
+        QPointF delta = end - start;
+
+        ulong timeDelta = te->timestamp() - touchBeginTimestamp;
+
+        if (qAbs(delta.x()) <= maxDisDelta && qAbs(delta.y()) <= maxDisDelta
+         && timeDelta < maxTimestampDelta) {
+            QMouseEvent *fakeMouseEvent = new QMouseEvent(QEvent::MouseButtonPress, start, Qt::LeftButton, nullptr, nullptr);
+            QApplication::sendEvent(viewport(), fakeMouseEvent);
+        }
+        return true;
+    }
+#else
     case QEvent::TouchEnd:
         return false;
+#endif
 #ifndef QT_NO_GESTURES
     case QEvent::Gesture:
     {
diff --git a/src/widgets/widgets/qabstractscrollarea.h b/src/widgets/widgets/qabstractscrollarea.h
index 6d0d8fa6e6..10bc3046a8 100644
--- a/src/widgets/widgets/qabstractscrollarea.h
+++ b/src/widgets/widgets/qabstractscrollarea.h
@@ -144,6 +144,8 @@ private:
 
     friend class QStyleSheetStyle;
     friend class QWidgetPrivate;
+
+    ulong touchBeginTimestamp;
 };
 
 #endif // QT_CONFIG(scrollarea)
