diff --git a/qtbase/src/plugins/platforms/xcb/qxcbconnection.cpp b/qtbase/src/plugins/platforms/xcb/qxcbconnection.cpp
index 05c0224..8fa9345 100644
--- a/qtbase/src/plugins/platforms/xcb/qxcbconnection.cpp
+++ b/qtbase/src/plugins/platforms/xcb/qxcbconnection.cpp
@@ -1687,8 +1687,13 @@ void QXcbConnection::processXcbEvents()
             continue;
         }
 
-        if (compressEvent(event, i, eventqueue))
-            continue;
+
+        /**
+         * (Krita) Do *NOT* compress any motion events! This is the
+         * most weird thing one can do for a painting application!
+         */
+        // if (compressEvent(event, i, eventqueue))
+        //     continue;
 
         bool accepted = false;
         if (clipboard()->processIncr())
