--- old/configure	2016-06-10 02:48:56.000000000 -0400
+++ new/configure	2016-09-21 02:26:34.000000000 -0400
@@ -548,7 +548,7 @@
         exit 2
     fi
 
-    if ! /usr/bin/xcrun -find xcrun >/dev/null 2>&1; then
+    if ! /usr/bin/xcrun -find xcodebuild >/dev/null 2>&1; then
         echo >&2
         echo "   Xcode not set up properly. You may need to confirm the license" >&2
         echo "   agreement by running /usr/bin/xcodebuild without arguments." >&2
