diff --git a/libde265/CMakeLists.txt b/libde265/CMakeLists.txt
index 48e3c80c..30c362d4 100644
--- a/libde265/CMakeLists.txt
+++ b/libde265/CMakeLists.txt
@@ -43,6 +43,8 @@ set (libde265_headers
   bitstream.h
   cabac.h
   configparam.h
+  de265.h
+  de265-version.h
   deblock.h
   decctx.h
   dpb.h
@@ -73,7 +75,7 @@ set (libde265_headers
   vui.h
 )
 
-if(MSVC)
+if(MSVC OR MINGW)
   set (libde265_sources
     ${libde265_sources}
     ../extra/win32cond.c
diff --git a/libde265/x86/CMakeLists.txt b/libde265/x86/CMakeLists.txt
index 0fd6fcfe..d6caf1d0 100644
--- a/libde265/x86/CMakeLists.txt
+++ b/libde265/x86/CMakeLists.txt
@@ -13,7 +13,11 @@ add_library(x86_sse OBJECT ${x86_sse_sources})
 set(sse_flags "")
 
 if(NOT MSVC)
-  set(sse_flags "${sse_flags} -msse4.1")
+  if(CMAKE_SIZEOF_VOID_P EQUAL 8)
+    set(sse_flags "${sse_flags} -msse4.1")
+  else(CMAKE_SIZEOF_VOID_P EQUAL 8)
+    set(sse_flags "${sse_flags} -msse2 -mssse3 -msse4.1")
+  endif(CMAKE_SIZEOF_VOID_P EQUAL 8)
 endif()
 
 set(X86_OBJECTS $<TARGET_OBJECTS:x86> $<TARGET_OBJECTS:x86_sse> PARENT_SCOPE)
