--- /home/sh_zam/workspace/build-krita-arm64-v8a/b/ext_heif/ext_libaom-prefix/src/ext_libaom/CMakeLists.txt	2023-05-26 23:00:57.037430105 +1000
+++ /home/sh_zam/workspace/tmp-build-krita/b/ext_heif/ext_libaom-prefix/src/ext_libaom/CMakeLists.txt	2023-05-31 01:43:29.529977035 +1000
@@ -212,6 +212,7 @@
 
 if(CONFIG_RUNTIME_CPU_DETECT AND ANDROID_NDK)
   include_directories(${ANDROID_NDK}/sources/android/cpufeatures)
+  list(APPEND AOM_SOURCES ${ANDROID_NDK}/sources/android/cpufeatures/cpu-features.c)
 endif()
 
 # Targets
