diff --git a/samples/CMakeLists.txt b/samples/CMakeLists.txt
index 0d9663d..c8bf137 100644
--- a/samples/CMakeLists.txt
+++ b/samples/CMakeLists.txt
@@ -92,9 +92,6 @@ if( EXPAT_FOUND )
 endif()
 
 # ******************************************************************************
-# connection test application
-add_executable(conntest conntest.cpp)
-list(APPEND APPLICATIONS conntest)
 
 if (EXIV2_ENABLE_WEBREADY)
     if( EXIV2_ENABLE_CURL )
@@ -110,11 +107,6 @@ if (EXIV2_ENABLE_WEBREADY)
     endif()
 endif()
 
-# ******************************************************************************
-# remotetest application
-add_executable(remotetest remotetest.cpp)
-list(APPEND APPLICATIONS remotetest)
-
 # ******************************************************************************
 foreach(application ${APPLICATIONS})
     target_link_libraries(${application} PRIVATE exiv2lib)
