From 647345ef8016141ac42a158885a24895f309b991 Mon Sep 17 00:00:00 2001
From: "L. E. Segovia" <amy@amyspark.me>
Date: Sun, 20 Dec 2020 23:01:29 +0000
Subject: [PATCH 4/4] Enable all input modes for Krita (plugin)

---
 src/Host/KritaPlugin/gmicqttoolplugin.cpp | 6 +++---
 1 file changed, 3 insertions(+), 3 deletions(-)

diff --git a/src/Host/KritaPlugin/gmicqttoolplugin.cpp b/src/Host/KritaPlugin/gmicqttoolplugin.cpp
index ad1e912..d62e76b 100644
--- a/src/Host/KritaPlugin/gmicqttoolplugin.cpp
+++ b/src/Host/KritaPlugin/gmicqttoolplugin.cpp
@@ -46,13 +46,13 @@ KritaGmicPlugin::KritaGmicPlugin(QObject *parent, const QVariantList &)
 
 int KritaGmicPlugin::launch(std::shared_ptr<KisImageInterface> i,
                             bool headless) {
-  disableInputMode(GmicQt::NoInput);
+  // disableInputMode(GmicQt::NoInput);
   // disableInputMode(GmicQt::Active);
   // disableInputMode(GmicQt::All);
   // disableInputMode(GmicQt::ActiveAndBelow);
   // disableInputMode(GmicQt::ActiveAndAbove);
-  disableInputMode(GmicQt::AllVisible);
-  disableInputMode(GmicQt::AllInvisible);
+  // disableInputMode(GmicQt::AllVisible);
+  // disableInputMode(GmicQt::AllInvisible);
 
   // disableOutputMode(GmicQt::InPlace);
   disableOutputMode(GmicQt::NewImage);
-- 
2.29.2

