From 4127ea5fdd9b8f058f49539c097c98735806c281 Mon Sep 17 00:00:00 2001
From: "L. E. Segovia" <amy@amyspark.me>
Date: Sun, 27 Jun 2021 02:08:49 +0000
Subject: [PATCH 9/9] Increase on-launch update timeout to 60s

---
 src/MainWindow.cpp | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/MainWindow.cpp b/src/MainWindow.cpp
index af2a8ff..d214d5f 100644
--- a/src/MainWindow.cpp
+++ b/src/MainWindow.cpp
@@ -1153,7 +1153,7 @@ void MainWindow::showEvent(QShowEvent * event)
   }
   const bool useNetwork = (ageLimit != INTERNET_NEVER_UPDATE_PERIODICITY);
   ui->progressInfoWidget->startFiltersUpdateAnimationAndShow();
-  Updater::getInstance()->startUpdate(ageLimit, 4, useNetwork);
+  Updater::getInstance()->startUpdate(ageLimit, 60, useNetwork);
 }
 
 void MainWindow::resizeEvent(QResizeEvent * e)
-- 
2.31.1.windows.1

