diff --git a/CMakeLists.txt b/CMakeLists.txt
index baf3946..c3f3ea4 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -698,7 +698,7 @@ endforeach ()
 if (BUILD_SHARED_LIBS)
   include(CheckCCompilerFlag)
   check_c_compiler_flag(-fPIC HAVE_FPIC)
-  if (HAVE_FPIC)
+  if (HAVE_FPIC AND NOT WIN32)
     add_definitions(-fPIC)
   endif ()
   if (WIN32)
@@ -791,7 +791,7 @@ if (GSL_INSTALL OR NOT DEFINED GSL_INSTALL)
     DESTINATION 
       lib/cmake/${PACKAGE_NAME}-${PACKAGE_VERSION} )
 endif ()
-if(NOT MSVC)
+if(NOT WIN32)
     target_compile_options(gsl PRIVATE -fPIC)
     target_compile_options(gslcblas PRIVATE -fPIC)
 endif()
