diff --git a/libheif/CMakeLists.txt b/libheif/CMakeLists.txt
index 59b4719..8fad622 100644
--- a/libheif/CMakeLists.txt
+++ b/libheif/CMakeLists.txt
@@ -107,6 +107,11 @@ if(AOM_ENCODER_FOUND OR AOM_DECODER_FOUND)
 
   include_directories(SYSTEM ${AOM_INCLUDE_DIR})
   target_link_libraries(heif PRIVATE ${AOM_LIBRARIES} ${AOM_LINKDIR})
+
+  find_package(Threads)
+  if (Threads_FOUND AND CMAKE_USE_PTHREADS_INIT)
+    target_link_libraries(heif PRIVATE Threads::Threads)
+  endif()
 endif()
 if(AOM_ENCODER_FOUND)
   target_compile_definitions(heif PRIVATE HAVE_AOM_ENCODER=1)
