From e5972ccd998a39efc562af6edb0a107b2b26ff37 Mon Sep 17 00:00:00 2001
From: Alonso Schaich <alonsoschaich@fastmail.fm>
Date: Thu, 24 Mar 2022 21:03:32 +0000
Subject: [PATCH 1/2] [fontconfig] Do not create symlinks (microsoft#23735)

---
 conf.d/link_confs.py | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/conf.d/link_confs.py b/conf.d/link_confs.py
index 52b8093..03608c6 100644
--- a/conf.d/link_confs.py
+++ b/conf.d/link_confs.py
@@ -4,6 +4,7 @@ import os
 import sys
 import argparse
 import platform
+import shutil
 
 if __name__=='__main__':
     parser = argparse.ArgumentParser()
@@ -32,7 +33,7 @@ if __name__=='__main__':
         except FileNotFoundError:
             pass
         try:
-            os.symlink(src, dst)
+            shutil.copyfile(src, dst)
         except NotImplementedError:
             # Not supported on this version of Windows
             break
-- 
2.32.0.windows.2

