From 25c277105642e601203f146bba782582f09bfd15 Mon Sep 17 00:00:00 2001
From: Sharaf Zaman <shzam@sdf.org>
Date: Sun, 21 May 2023 01:28:34 +1000
Subject: [PATCH 3/3] Do not build bin/melt on Android.

---
 src/melt/CMakeLists.txt | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/melt/CMakeLists.txt b/src/melt/CMakeLists.txt
index 75a87e31..7e3e397f 100644
--- a/src/melt/CMakeLists.txt
+++ b/src/melt/CMakeLists.txt
@@ -6,7 +6,7 @@ target_link_libraries(melt PRIVATE mlt Threads::Threads)
 
 target_compile_definitions(melt PRIVATE VERSION="${MLT_VERSION}")
 
-if(TARGET PkgConfig::sdl2)
+if(TARGET PkgConfig::sdl2 AND NOT ANDROID)
     target_link_libraries(melt PRIVATE PkgConfig::sdl2)
     target_compile_definitions(melt PRIVATE HAVE_SDL2)
     if(MINGW)
-- 
2.39.2

