From 1c110d252c65d51752adbd323f35c4d99d3e996c Mon Sep 17 00:00:00 2001
From: "L. E. Segovia" <amy@amyspark.me>
Date: Thu, 10 Mar 2022 10:38:18 -0300
Subject: [PATCH] Publish FreeType dependencies

---
 CMakeLists.txt | 30 +++++++++++++++++-------------
 1 file changed, 17 insertions(+), 13 deletions(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index 4335d66dc..66673ea39 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -496,30 +496,34 @@ set(PKG_CONFIG_REQUIRED_PRIVATE "")
 set(PKG_CONFIG_LIBS_PRIVATE "")
 
 if (ZLIB_FOUND)
-  target_link_libraries(freetype PRIVATE ${ZLIB_LIBRARIES})
-  target_include_directories(freetype PRIVATE ${ZLIB_INCLUDE_DIRS})
+  target_link_libraries(freetype PUBLIC ${ZLIB_LIBRARIES})
+  target_include_directories(freetype PUBLIC ${ZLIB_INCLUDE_DIRS})
   list(APPEND PKG_CONFIG_REQUIRED_PRIVATE "zlib")
 endif ()
 if (BZIP2_FOUND)
-  target_link_libraries(freetype PRIVATE ${BZIP2_LIBRARIES})
-  target_include_directories(freetype PRIVATE ${BZIP2_INCLUDE_DIR}) # not BZIP2_INCLUDE_DIRS
-  list(APPEND PKG_CONFIG_LIBS_PRIVATE "-lbz2")
+  target_link_libraries(freetype PUBLIC ${BZIP2_LIBRARIES})
+  target_include_directories(freetype PUBLIC ${BZIP2_INCLUDE_DIR}) # not BZIP2_INCLUDE_DIRS
+  if (PC_BZIP2_FOUND)
+    list(APPEND PKG_CONFIG_REQUIRED_PRIVATE "bzip2")
+  else ()
+    list(APPEND PKG_CONFIG_LIBS_PRIVATE "-lbz2")
+  endif ()
 endif ()
 if (PNG_FOUND)
-  target_link_libraries(freetype PRIVATE ${PNG_LIBRARIES})
-  target_compile_definitions(freetype PRIVATE ${PNG_DEFINITIONS})
-  target_include_directories(freetype PRIVATE ${PNG_INCLUDE_DIRS})
+  target_link_libraries(freetype PUBLIC ${PNG_LIBRARIES})
+  target_compile_definitions(freetype PUBLIC ${PNG_DEFINITIONS})
+  target_include_directories(freetype PUBLIC ${PNG_INCLUDE_DIRS})
   list(APPEND PKG_CONFIG_REQUIRED_PRIVATE "libpng")
 endif ()
 if (HarfBuzz_FOUND)
-  target_link_libraries(freetype PRIVATE ${HarfBuzz_LIBRARY})
-  target_include_directories(freetype PRIVATE ${HarfBuzz_INCLUDE_DIRS})
+  target_link_libraries(freetype PUBLIC ${HarfBuzz_LIBRARY})
+  target_include_directories(freetype PUBLIC ${HarfBuzz_INCLUDE_DIRS})
   list(APPEND PKG_CONFIG_REQUIRED_PRIVATE "harfbuzz >= ${HARFBUZZ_MIN_VERSION}")
 endif ()
 if (BROTLIDEC_FOUND)
-  target_link_libraries(freetype PRIVATE ${BROTLIDEC_LIBRARIES})
-  target_compile_definitions(freetype PRIVATE ${BROTLIDEC_DEFINITIONS})
-  target_include_directories(freetype PRIVATE ${BROTLIDEC_INCLUDE_DIRS})
+  target_link_libraries(freetype PUBLIC ${BROTLIDEC_LIBRARIES})
+  target_compile_definitions(freetype PUBLIC ${BROTLIDEC_DEFINITIONS})
+  target_include_directories(freetype PUBLIC ${BROTLIDEC_INCLUDE_DIRS})
   list(APPEND PKG_CONFIG_REQUIRED_PRIVATE "libbrotlidec")
 endif ()
 
-- 
2.32.0.windows.2

