From f7c103e5c21a750c8560d7bc6fddb89d66a1188d Mon Sep 17 00:00:00 2001
From: "L. E. Segovia" <amy@amyspark.me>
Date: Fri, 19 Feb 2021 11:27:33 +0000
Subject: [PATCH 2/4] Quick check for SSE support

---
 src/CMakeLists.txt | 19 +++++++++++++++++++
 1 file changed, 19 insertions(+)

diff --git a/src/CMakeLists.txt b/src/CMakeLists.txt
index b83ea07..9bccbc6 100644
--- a/src/CMakeLists.txt
+++ b/src/CMakeLists.txt
@@ -17,6 +17,25 @@ if(WIN32)
     endif(MSVC)
 endif(WIN32)
 
+include(CheckCXXSourceCompiles)
+check_cxx_source_compiles ("
+    #include <immintrin.h>
+    int main ()
+    {
+        __m128d a, b;
+        double vals[2] = {0};
+        a = _mm_loadu_pd (vals);
+        b = _mm_add_pd (a,a);
+        _mm_storeu_pd (vals,b);
+        return (0);
+    }"
+    HAVE_SSE2)
+
+if (NOT HAVE_SSE2)
+    message(STATUS "Disabling SSE optimizations, as the target doesn't support them")
+    add_compile_definitions(-DCMS_DONT_USE_SSE2=1)
+endif()
+
 set(lcms_SRCS
     cmscnvrt.c
     cmserr.c
-- 
2.32.0.windows.2

