diff --git a/src/raqm.c b/src/raqm.c
index 2e7ac71..7bf3ef9 100644
--- a/src/raqm.c
+++ b/src/raqm.c
@@ -1305,7 +1305,9 @@ raqm_layout (raqm_t *rq)
 static uint32_t
 _raqm_u32_to_u8_index (raqm_t   *rq,
                        uint32_t  index);
-
+static uint32_t
+_raqm_u32_to_u16_index (raqm_t   *rq,
+                       uint32_t  index);
 /**
  * raqm_get_glyphs:
  * @rq: a #raqm_t.
@@ -1403,6 +1405,10 @@ raqm_get_glyphs (raqm_t *rq,
       RAQM_TEST (" %02d", rq->glyphs[i].cluster);
     RAQM_TEST ("\n");
 #endif
+  } else if (rq->text_utf16) {
+    for (size_t i = 0; i < count; i++)
+      rq->glyphs[i].cluster = _raqm_u32_to_u16_index (rq,
+                                                     rq->glyphs[i].cluster);
   }
   return rq->glyphs;
 }
@@ -2301,7 +2307,7 @@ _raqm_u8_to_u32_index (raqm_t   *rq,
 static size_t
 _raqm_count_codepoint_utf16_short (uint32_t chr)
 {
-  if (chr > 0x010000)
+  if (chr > 0x00010000)
     return 2;
   else
     return 1;
