From 4028cbbc95a23a69fff02ab91f5589ead75f657f Mon Sep 17 00:00:00 2001
From: Dmitry Kazakov <dimula73@gmail.com>
Date: Tue, 23 May 2023 14:33:42 +0300
Subject: [PATCH] Fix build on Linux

---
 meson.build | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/meson.build b/meson.build
index a2eeacf17..0468cd1e0 100644
--- a/meson.build
+++ b/meson.build
@@ -1137,7 +1137,7 @@ endif
 
 # Almost every platform uses pthreads.
 if features.get('multithread', false)
-	threads_dep = c.find_library('threads', required: false)
+	threads_dep = dependency('threads')
 	if c.compiles('''
 		#include <pthread.h>
 		#include <stddef.h>
-- 
2.25.1

