From 500117153b23d2b7b7052bf34580d4c734bcb087 Mon Sep 17 00:00:00 2001
From: Alvin Wong <alvinhochun@gmail.com>
Date: Sat, 12 Mar 2022 21:19:11 +0800
Subject: [PATCH 14/17] CMake: Do not exclude EGL headers

---
 CMakeLists.txt | 10 +++++-----
 1 file changed, 5 insertions(+), 5 deletions(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index 08fcc235c..e62471e07 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -737,11 +737,11 @@ if(NOT DISABLE_INSTALL_HEADERS)
         PATTERN "*.h"
         PATTERN "*.inc"
         PATTERN "CL" EXCLUDE
-        PATTERN "GLSLANG" EXCLUDE
-        PATTERN "egl.h" EXCLUDE
-        PATTERN "eglext.h" EXCLUDE
-        PATTERN "eglplatform.h" EXCLUDE
-        PATTERN "KHR" EXCLUDE
+        #PATTERN "GLSLANG" EXCLUDE
+        #PATTERN "egl.h" EXCLUDE
+        #PATTERN "eglext.h" EXCLUDE
+        #PATTERN "eglplatform.h" EXCLUDE
+        #PATTERN "KHR" EXCLUDE
         PATTERN "WGL" EXCLUDE
         PATTERN "export.h" EXCLUDE
     )
-- 
2.24.1.windows.2

