From 906e05f3fb14b84fe23bc09b67ec2b624536a168 Mon Sep 17 00:00:00 2001
From: Ivan Yossi <ghevan@gmail.com>
Date: Thu, 7 Dec 2023 01:40:41 -0600
Subject: [PATCH] restrict sip version

---
 setup.py | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/setup.py b/setup.py
index 3c8d47f..feb53ef 100644
--- a/setup.py
+++ b/setup.py
@@ -49,7 +49,7 @@ setup(
         version=version,
         license='SIP',
         python_requires='>=3.7',
-        install_requires=['packaging', 'sip >=6.3, <7'],
+        install_requires=['packaging', 'sip >=6.3, <6.8'],
         packages=find_packages(),
         package_data={
             'pyqtbuild.bundle': ['dlls/*/*', 'qt_wheel_distinfo/*'],
-- 
2.37.1 (Apple Git-137.1)

