From 3e6b95df0130e2e1320c1bbe1dbce8ea35fed07d Mon Sep 17 00:00:00 2001
From: "L. E. Segovia" <amy@amyspark.me>
Date: Thu, 2 Sep 2021 01:02:57 +0000
Subject: [PATCH 2/2] Add additional flavor of MinGW

---
 pyqtbuild/builder.py | 6 ++++--
 1 file changed, 4 insertions(+), 2 deletions(-)

diff --git a/pyqtbuild/builder.py b/pyqtbuild/builder.py
index 0444c7f..eb51a06 100644
--- a/pyqtbuild/builder.py
+++ b/pyqtbuild/builder.py
@@ -334,8 +334,10 @@ class QmakeBuilder(Builder):
         """ Return the name of a valid make program. """
 
         if self.project.py_platform == 'win32':
-            if 'g++' in self.spec:
-                make = 'make'
+            if self.spec == 'win32-g++' or self.spec == 'win32-clang-g++':
+                make = 'mingw32-make'
+                if self._find_exe(make) is None:
+                    make = 'make'
             else:
                 make = 'nmake'
         else:
-- 
2.37.1.windows.1

