--- a/configure.py
+++ b/configure.py
@@ -965,7 +965,7 @@ class TargetConfiguration:
         # it where it is.
         if not self.static:
             pro_lines.extend(['win32 {',
-                    '    LIBS += ' + self.get_pylib_link_arguments(name=False),
+                    '    LIBS += ' + self.get_pylib_link_arguments(),
                     '}'])
 
     @staticmethod
@@ -1977,7 +1977,7 @@ def run_make(target_config, verbose, exe, makefile_name):
 
     remove_file(platform_exe)
 
-    args = [make, '-f', makefile_name]
+    args = [make, '-f', makefile_name, 'CXXFLAGS=-D_hypot=hypot']
 
     if makefile_target != '':
         args.append(makefile_target)
