From 01a8979dc75293a8bbc038410dcc9e1569c61464 Mon Sep 17 00:00:00 2001
From: Ivan Yossi <ghevan@gmail.com>
Date: Mon, 25 Sep 2023 12:44:12 -0600
Subject: [PATCH] fix linker detection on Xcode 15

---
 mesonbuild/linkers/detect.py | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/mesonbuild/linkers/detect.py b/mesonbuild/linkers/detect.py
index 97e770c..03e80b6 100644
--- a/mesonbuild/linkers/detect.py
+++ b/mesonbuild/linkers/detect.py
@@ -218,7 +218,7 @@ def guess_nix_linker(env: 'Environment', compiler: T.List[str], comp_class: T.Ty
         mlog.debug(f'linker stderr:\n{newerr}')
 
         for line in newerr.split('\n'):
-            if 'PROJECT:ld' in line:
+            if 'PROJECT:ld' or 'PROJECT:dyld' in line:
                 v = line.split('-')[1]
                 break
         else:
-- 
2.37.1 (Apple Git-137.1)

