Cherry picked from libjxl 3cac2874bb125c901f1703065f658e32e7c85cee

diff --git a/lib/jxl/cms/color_encoding_cms.h b/lib/jxl/cms/color_encoding_cms.h
index 81b32897..0ad1cbb4 100644
--- a/lib/jxl/cms/color_encoding_cms.h
+++ b/lib/jxl/cms/color_encoding_cms.h
@@ -521,7 +521,6 @@ struct ColorEncoding {
                                                 new_icc.data(), new_icc.size(),
                                                 &external, &new_cmyk));
     cmyk = new_cmyk;
-    if (cmyk) return true;
     JXL_RETURN_IF_ERROR(FromExternal(external));
     icc = std::move(new_icc);
     return true;
