From 4470988d9aea2cf48bdaf8e88f06f5868e9c7fdd Mon Sep 17 00:00:00 2001
From: "L. E. Segovia" <amy@amyspark.me>
Date: Thu, 18 Feb 2021 15:31:16 +0000
Subject: [PATCH 2/2] Use CMake install prefix

---
 CMakeLists.txt | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index ea912d5..6f89e4a 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -107,8 +107,8 @@ SET(DEMOSAIC_PACK_GPL2_PATH "${CMAKE_CURRENT_SOURCE_DIR}/${DEMOSAIC_PACK_GPL2_RP
 SET(DEMOSAIC_PACK_GPL3_PATH "${CMAKE_CURRENT_SOURCE_DIR}/${DEMOSAIC_PACK_GPL3_RPATH}")
 SET(RAWSPEED_PATH           "${CMAKE_CURRENT_SOURCE_DIR}/${RAWSPEED_RPATH}")
 
-SET(INSTALL_CMAKE_MODULE_PATH  "share/libraw/cmake"  CACHE STRING
-    "Path to install cmake module              (default=share/libraw/cmake)")
+SET(INSTALL_CMAKE_MODULE_PATH  "${CMAKE_INSTALL_PREFIX}/lib/cmake/libraw"  CACHE STRING
+    "Path to install cmake module              (default=CMAKE_INSTALL_PREFIX/lib/cmake/libraw)")
 
 # ==================================================================================================
 # General definitions rules
-- 
2.29.2.windows.1

