diff --git a/3rdparty/kdchart/kdablibfakes/src/KDABLibFakes.h b/3rdparty/kdchart/kdablibfakes/src/KDABLibFakes.h
index 0bfacab..bcb6bf0 100644
--- a/3rdparty/kdchart/kdablibfakes/src/KDABLibFakes.h
+++ b/3rdparty/kdchart/kdablibfakes/src/KDABLibFakes.h
@@ -56,8 +56,8 @@
 #define ISNAN(x) std::isnan(x)
 #define ISINF(x) std::isinf(x)
 #else
-#define ISNAN(x) isnan(x)
-#define ISINF(x) isinf(x)
+#define ISNAN(x) std::isnan(x)
+#define ISINF(x) std::isinf(x)
 #endif
 
 
