From c8cbb1119bbd9749b44aafda40f50082951d098d Mon Sep 17 00:00:00 2001
From: "L. E. Segovia" <amy@amyspark.me>
Date: Fri, 1 Oct 2021 03:05:18 +0000
Subject: [PATCH 3/4] ExprEditor: replace index for line/column when reporting
 errors

---
 src/KSeExprUI/ExprEditor.cpp | 12 ++++++++----
 1 file changed, 8 insertions(+), 4 deletions(-)

diff --git a/src/KSeExprUI/ExprEditor.cpp b/src/KSeExprUI/ExprEditor.cpp
index d5810b3..f90c7f6 100644
--- a/src/KSeExprUI/ExprEditor.cpp
+++ b/src/KSeExprUI/ExprEditor.cpp
@@ -213,16 +213,15 @@ void ExprEditor::appendStr(const QString &str)
 
 void ExprEditor::addError(const int startPos, const int endPos, const QString &error)
 {
-    QString message = tr("(%1, %2): %3").arg(startPos).arg(endPos).arg(error);
-    auto *item = new QListWidgetItem(message, errorWidget);
-    item->setData(Qt::UserRole, startPos);
-    item->setData(Qt::UserRole + 1, endPos);
+    int startLine, startCol;
     errorWidget->setHidden(false);
 
     // Underline error
     QTextCursor cursor = exprTe->textCursor();
     cursor.movePosition(QTextCursor::Start, QTextCursor::MoveAnchor);
     cursor.movePosition(QTextCursor::Right, QTextCursor::MoveAnchor, startPos);
+    startLine = cursor.blockNumber();
+    startCol = cursor.columnNumber();
     cursor.movePosition(QTextCursor::Right, QTextCursor::KeepAnchor, endPos - startPos + 1);
     QList<QTextEdit::ExtraSelection> extraSelections = exprTe->extraSelections();
     QTextEdit::ExtraSelection selection;
@@ -233,6 +232,11 @@ void ExprEditor::addError(const int startPos, const int endPos, const QString &e
     extraSelections.append(selection);
     exprTe->setExtraSelections(extraSelections);
 
+    QString message = tr("(%1, %2): %3").arg(startLine).arg(startCol).arg(error);
+    auto *item = new QListWidgetItem(message, errorWidget);
+    item->setData(Qt::UserRole, startPos);
+    item->setData(Qt::UserRole + 1, endPos);
+
     // errorWidget has its height fixed -- amyspark
     // TODO: fix to not use count lines and compute heuristic of 25 pixels per line!
     // const char* c = error.c_str();
-- 
2.32.0.windows.2

