diff --git a/Include/pyport.h b/Include/pyport.h
index 66e00d4..89f7c0b 100644
--- a/Include/pyport.h
+++ b/Include/pyport.h
@@ -688,6 +688,12 @@ extern pid_t forkpty(int *, char *, struct termios *, struct winsize *);
 #endif
 
 #ifdef _PY_PORT_CTYPE_UTF8_ISSUE
+#ifndef __cplusplus
+ /* The workaround below is unsafe in C++ because
+  * the <locale> defines these symbols as real functions,
+  * with a slightly different signature.
+  * See issue #10910
+  */
 #include <ctype.h>
 #include <wctype.h>
 #undef isalnum
@@ -705,6 +711,7 @@ extern pid_t forkpty(int *, char *, struct termios *, struct winsize *);
 #undef toupper
 #define toupper(c) towupper(btowc(c))
 #endif
+#endif
 
 
 /* Declarations for symbol visibility.
