diff --git a/CMakeLists.txt b/CMakeLists.txt
index 9da4953..4fd8c7b 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -272,7 +272,7 @@ set (gmic_qt_SRCS
   src/FilterParameters/ChoiceParameter.h
   src/FilterParameters/ColorParameter.h
   src/FilterParameters/ConstParameter.h
-  src/FilterParameters/CustomDoubleSpinbox.h
+  src/FilterParameters/CustomDoubleSpinBox.h
   src/FilterParameters/FileParameter.h
   src/FilterParameters/FilterParametersWidget.h
   src/FilterParameters/FloatParameter.h
@@ -344,7 +344,7 @@ set (gmic_qt_SRCS
   src/FilterParameters/ChoiceParameter.cpp
   src/FilterParameters/ColorParameter.cpp
   src/FilterParameters/ConstParameter.cpp
-  src/FilterParameters/CustomDoubleSpinbox.cpp
+  src/FilterParameters/CustomDoubleSpinBox.cpp
   src/FilterParameters/FileParameter.cpp
   src/FilterParameters/FilterParametersWidget.cpp
   src/FilterParameters/FloatParameter.cpp
