diff --git a/src/CImg.h b/src/CImg.h
index 2b7a53b..886daf7 100644
--- a/src/CImg.h
+++ b/src/CImg.h
@@ -3205,7 +3205,7 @@ namespace cimg_library_suffixed {
 #if defined(cimg_use_fftw3) && !defined(cimg_use_fftw3_singlethread)
     struct FFTW3_static {
       FFTW3_static() {
-        fftw_init_threads();
+        //fftw_init_threads();
       }
     }; // struct FFTW3_static { ...
     static FFTW3_static _FFTW3_static;
