diff --git a/cmake/OptimizeForArchitecture.cmake b/cmake/OptimizeForArchitecture.cmake
index 075956c..b4c47b7 100644
--- a/cmake/OptimizeForArchitecture.cmake
+++ b/cmake/OptimizeForArchitecture.cmake
@@ -70,12 +70,13 @@ macro(AutodetectHostArchitecture)
       string(REGEX REPLACE ".*model[ \t]*:[ \t]+([a-zA-Z0-9_-]+).*" "\\1" _cpu_model "${_cpuinfo}")
       string(REGEX REPLACE ".*flags[ \t]*:[ \t]+([^\n]+).*" "\\1" _cpu_flags "${_cpuinfo}")
    elseif(CMAKE_SYSTEM_NAME STREQUAL "Darwin")
+      message(STATUS "${CMAKE_SYSTEM_PROCESSOR} --  ${CMAKE_HOST_SYSTEM_PROCESSOR}")
-      exec_program("/usr/sbin/sysctl -n machdep.cpu.vendor machdep.cpu.model machdep.cpu.family machdep.cpu.features" OUTPUT_VARIABLE _sysctl_output_string)
+      exec_program("/usr/sbin/sysctl -n machdep.cpu.family machdep.cpu.features" OUTPUT_VARIABLE _sysctl_output_string)
       string(REPLACE "\n" ";" _sysctl_output ${_sysctl_output_string})
-      list(GET _sysctl_output 0 _vendor_id)
-      list(GET _sysctl_output 1 _cpu_model)
-      list(GET _sysctl_output 2 _cpu_family)
-      list(GET _sysctl_output 3 _cpu_flags)
+      set(_vendor_id "GenuineIntel")
+      set(_cpu_model 13)
+      list(GET _sysctl_output 0 _cpu_family)
+      list(GET _sysctl_output 1 _cpu_flags)

       string(TOLOWER "${_cpu_flags}" _cpu_flags)
       string(REPLACE "." "_" _cpu_flags "${_cpu_flags}")
