From aeddd16fcd39e78d8208c78c17f06d24632b7560 Mon Sep 17 00:00:00 2001
From: Alvin Wong <alvinhochun@gmail.com>
Date: Thu, 10 Mar 2022 16:26:23 +0800
Subject: [PATCH 12/17] Fix Clang MinGW target build

---
 src/libANGLE/CLRefPointer.h | 8 ++++----
 1 file changed, 4 insertions(+), 4 deletions(-)

diff --git a/src/libANGLE/CLRefPointer.h b/src/libANGLE/CLRefPointer.h
index 6ea46978f..9ec024ac4 100644
--- a/src/libANGLE/CLRefPointer.h
+++ b/src/libANGLE/CLRefPointer.h
@@ -104,25 +104,25 @@ void swap(RefPointer<T> &left, RefPointer<T> &right)
 }
 
 template <typename T>
-bool operator==(const RefPointer<T> &ptr, nullptr_t) noexcept
+bool operator==(const RefPointer<T> &ptr, std::nullptr_t) noexcept
 {
     return ptr.get() == nullptr;
 }
 
 template <typename T>
-bool operator==(nullptr_t, const RefPointer<T> &ptr) noexcept
+bool operator==(std::nullptr_t, const RefPointer<T> &ptr) noexcept
 {
     return ptr.get() == nullptr;
 }
 
 template <typename T>
-bool operator!=(const RefPointer<T> &ptr, nullptr_t) noexcept
+bool operator!=(const RefPointer<T> &ptr, std::nullptr_t) noexcept
 {
     return ptr.get() != nullptr;
 }
 
 template <typename T>
-bool operator!=(nullptr_t, const RefPointer<T> &ptr) noexcept
+bool operator!=(std::nullptr_t, const RefPointer<T> &ptr) noexcept
 {
     return ptr.get() != nullptr;
 }
-- 
2.24.1.windows.2

