From 1e7bc9d8d907a78c96d8b52d22e36348327fd643 Mon Sep 17 00:00:00 2001
From: Sharaf Zaman <shzam@sdf.org>
Date: Fri, 30 Sep 2022 08:03:02 +0000
Subject: [PATCH] Support Android NDK r22b

---
 build-android.sh | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/build-android.sh b/build-android.sh
index ae5a4fd..d302d1b 100755
--- a/build-android.sh
+++ b/build-android.sh
@@ -398,7 +398,7 @@ case "$NDK_RN" in
 		TOOLSET=clang
 		CONFIG_VARIANT=ndk19
 		;;
-	"23.0"|"23.1"|"25.0"|"25.1")
+	"22.1"|"23.0"|"23.1"|"25.0"|"25.1")
 		TOOLCHAIN=${TOOLCHAIN:-llvm}
 		CXXPATH=$AndroidNDKRoot/toolchains/${TOOLCHAIN}/prebuilt/${PlatformOS}-x86_64/bin/clang++
 		TOOLSET=clang
@@ -419,7 +419,7 @@ if [ -z "${ARCHLIST}" ]; then
 
     case "$NDK_RN" in
       # NDK 17+: Support for ARMv5 (armeabi), MIPS, and MIPS64 has been removed.
-      "17.1"|"17.2"|"18.0"|"18.1"|"19.0"|"19.1"|"19.2"|"20.0"|"20.1"|"21.0"|"21.1"|"21.2"|"21.3"|"21.4"|"23.0"|"23.1"|"25.0"|"25.1")
+      "17.1"|"17.2"|"18.0"|"18.1"|"19.0"|"19.1"|"19.2"|"20.0"|"20.1"|"21.0"|"21.1"|"21.2"|"21.3"|"21.4"|"22.1"|"23.0"|"23.1"|"25.0"|"25.1")
         ARCHLIST="arm64-v8a armeabi-v7a x86 x86_64"
         ;;
       *)
-- 
2.37.1.windows.1

