--- a/IlmImf/ImfSystemSpecific.h
+++ b/IlmImf/ImfSystemSpecific.h
@@ -54,7 +54,7 @@
 
 #ifdef IMF_HAVE_SSE2
 
-#ifdef __GNUC__
+#if defined(__GNUC__) && !defined(__MINGW__) && !defined(__MINGW32__) && !defined(__MINGW64__)
 // Causes issues on certain gcc versions
 //#define EXR_FORCEINLINE inline __attribute__((always_inline))
 #define EXR_FORCEINLINE inline
@@ -73,7 +73,7 @@
     free(ptr);
 }
 
-#elif defined _MSC_VER
+#elif defined(_MSC_VER) || defined(__MINGW__) || defined(__MINGW32__) || defined(__MINGW64__)
 
 #define EXR_FORCEINLINE __forceinline
 #define EXR_RESTRICT __restrict
--- a/IlmImf/ImfOptimizedPixelReading.h
+++ b/IlmImf/ImfOptimizedPixelReading.h
@@ -70,7 +70,7 @@
 bool
 isPointerSSEAligned (const void* EXR_RESTRICT pPointer)
 {
-    unsigned long trailingBits = ((unsigned long)pPointer) & 15;
+    unsigned long trailingBits = ((unsigned size_t)pPointer) & 15;
     return trailingBits == 0;
 }
 
